charts/redpanda/templates/statefulset.yaml

```diff
@@ -300,6 +314,18 @@ spec:
             path: {{ .Values.storage.hostPath | quote }}
 {{- else }}
           emptyDir: {}
+{{- end }}
+{{- if and (or (not (empty .Values.license_key)) (not (empty .Values.license_secret_ref)) .Values.storage.tieredConfig.cloud_storage_enabled) (hasKey .Values.storage.tieredConfig "cloud_storage_cache_directory") (not (empty .Values.storage.tieredConfig.cloud_storage_cache_directory)) }}
+        - name: tiered-storage-dir
+{{- if .Values.storage.tieredStoragePersistentVolume.enabled }}
+          persistentVolumeClaim:
+            claimName: tiered-storage-dir
+{{- else if .Values.storage.tieredStorageHostPath }}
+          hostPath:
+            path: {{ .Values.storage.tieredStorageHostPath | quote }}
+{{- else }}
+          emptyDir: {}
```

Comment:
`emptyDir` is a risky default here.  The tiered storage cache size is currently 20GiB (it's not very smart, the assumption is that the thing deploying will have opinions), and if you run that on a platform that uses 1GB emptyDirs then it could well end up asserting out and/or misbehaving.

We do also use that as the default for redpanda data dirs, but: I think people are more likely to intuitively understand that using emptyDir for their data dirs isn't going to work, whereas for a cache dir they might reasonably think this is a usable configuration.

Reply:
> emptyDir is a risky default here. The tiered storage cache size is currently 20GiB (it's not very smart, the assumption is that the thing deploying will have opinions), and if you run that on a platform that uses 1GB emptyDirs then it could well end up asserting out and/or misbehaving.

True, but I'm about to remove the explicit default and remove the comments, so where I should mention this caveat?

> We do also use that as the default for redpanda data dirs, but: I think people are more likely to intuitively understand that using emptyDir for their data dirs isn't going to work, whereas for a cache dir they might reasonably think this is a usable configuration.

The nice think about kubernetes is if user configure `emptyDir` for tiered storage cache it could end up running out of disk space. This event is reported by kubelet and node evicts some pods. If user create auto scaling node group, then VM should be in degraded state, which would trigger new instance creation. During that time Redpanda cluster will be in degraded state. Redpanda would picked Redpanda ID from data volume if it's backed by persistent disk, or generate new ID. Helm doesn't have ways to detect that Redpanda lost whole data folder, so no decommission of Redpanda node/broker would happen. That way, I would argue, we can not prevent from miss configuration. I'm not sure if documentation would be enough to prevent this. 

----

I did more and found emptyDir can have sizeLimit property set to prevent from your scenario. I will add this.

REF: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#emptydirvolumesource-v1-core